### PR TITLE
GCP-2 Global palette frontend implementation

### PR DESCRIPTION
**Pull request by patilvikasj**
### Description
Global Palette front end implementation

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/class-astra-dynamic-css.php [hunk 1]
```diff
@@ -3477,5 +3479,24 @@ public static function load_cart_static_css() {
...
+		 *
+		 * @since x.x.x
+		 * @return array palette style array.
+		 */
+		public static function generate_global_palette_style() {
+
+			$palette       = astra_get_option( 'global-color-palette' );
+			$palette_style = array();
+
+			foreach ( $palette['palette'] as $key => $color ) {
```
Comment (rohitp-bsf) [inline inc/class-astra-dynamic-css.php:3494]: Can you pull out `$palette['palette']` like `$palette =  isset( $palette['palette'] ) ? $palette['palette'] : array() ` for safe side?

File: inc/class-astra-dynamic-css.php [hunk 2]
```diff
@@ -3477,5 +3479,24 @@ public static function load_cart_static_css() {
...
+				$palette_key                   = '--ast-global-' . $key;
```
Comment (rohitp-bsf) [inline inc/class-astra-dynamic-css.php:3495]: It would be good if we can use const for '--ast-global-'.

**Review by rohitp-bsf — CHANGES_REQUESTED** [review]

File: inc/class-astra-dynamic-css.php (hunk 1 above)
Comment (patilvikasj) [reply inc/class-astra-dynamic-css.php]: Done.

File: inc/class-astra-dynamic-css.php (hunk 2 above)
Comment (patilvikasj) [reply inc/class-astra-dynamic-css.php]: We have now added a helper function to get the palette variable prefix.

**Review by rohitp-bsf — APPROVED** [review]

File: inc/class-astra-global-palette.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	 * Generate palette CSS required to display on front end.
+	 *
+	 * @since x.x.x
+	 * @return array palette style array.
+	 */
+	public static function generate_global_palette_style() {
+		$palette_data  = astra_get_option( 'global-color-palette' );
+		$palette_style = array();
+
+		$palette = isset( $palette_data['palette'] ) ? $palette_data['palette'] : '';
```
Comment (premanshup) [inline inc/class-astra-global-palette.php:41]:
```suggestion

```
`empty( $palette )` internally checks `isset( $palette )`

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 4]
```diff
@@ -44,78 +40,47 @@ const ColorPaletteComponent = (props) => {
...
+			</div>
 		</>
 	);
 
 	const updatePaletteVariables = (e) => {
-		props.control.setPaletteVariables(e.detail.palette);
+		props.control.setPaletteVariables(e.detail.data.palette);
 	};
 
 	document.addEventListener(
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:86]: @patilvikasj I think custom event can be removed along with https://github.com/brainstormforce/astra/pull/2580/files#diff-2c13e8e5e8d439689c38a56849e0a95a9f5d3de49ef8548e1760719c637ca31fR105. Please check.

**Review by premanshup — CHANGES_REQUESTED** [review]

**Pull request closed without merging**